tensorflow/lite/delegates/gpu/metal_delegate.mm

```diff
@@ -221,7 +226,12 @@ kernel void ComputeFunction(device int* output_buffer [[buffer(0)]],
     for (auto& output : graph_outputs_) {
       if (output.tensor_id == tensor_index) {
         if (in_out_tensors_[output.id]->GetBufferHandle() != buffer) {
-          RETURN_IF_ERROR(in_out_tensors_[output.id]->SetBufferHandle(buffer));
+          if ([in_out_tensors_[output.id]->GetBufferHandle() length] == [buffer length]) {
+            RETURN_IF_ERROR(in_out_tensors_[output.id]->SetBufferHandle(buffer));
+            output.is_bphwc4 = true;
+          } else {
+            in_out_bhwc_f32_buffers_[output.id] = buffer;
```

Comment:
Does this work? I think output buffer is usually created by delegate. I'm not sure how non phwc4 output will work.

Reply:
Again, this the case when you use `TFLGpuDelegateBindMetalBufferToTensor`. Using this method, you can bind external output buffer.

Reply:
If assigning to `in_out_bhwc_f32_buffers_` non_pbwc4 will be outputed using this code here. https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/delegates/gpu/metal_delegate.mm#L522

Reply:
Hi @terryheo Can you please assist on above comments from @jongkweh. Thank you!